internal/controller/helmchart_controller.go

```diff
@@ -760,8 +762,10 @@ func (r *HelmChartReconciler) buildFromTarballArtifact(ctx context.Context, obj
 
 	// Configure builder options, including any previously cached chart
 	opts := chart.BuildOptions{
-		ValuesFiles: obj.GetValuesFiles(),
-		Force:       obj.Generation != obj.Status.ObservedGeneration,
+		ValuesFiles:              obj.GetValuesFiles(),
+		ObservedValuesFiles:      obj.Status.ObservedValuesFiles,
+		IgnoreMissingValuesFiles: obj.Spec.IgnoreMissingValuesFiles,
+		Force:                    obj.Generation != obj.Status.ObservedGeneration,
 	}
 	if artifact := obj.Status.Artifact; artifact != nil {
```

Comment:
I would set it only if we retrieve a cached artifact, this should be fine as we use it only if we retrieve a cached artifact. Like

```go
	if artifact := obj.GetArtifact(); artifact != nil {
		opts.CachedChart = r.Storage.LocalPath(*artifact)
		opts.CachedChartValuesFiles = obj.Status.ObservedValuesFiles
	}
```